### PR TITLE
Port Azure CI jobs to new GitHub Actions workflow

### PR DESCRIPTION
**Pull request by kinke**

### CODE REVIEW

**Comment by ibuclaw** [comment]
@kinke who's going to be using this?

**Comment by Geod24** [comment]
Looks good! Can you rebase (and apply the diff from HEAD) so we can merge ?

File: .github/workflows/build_windows.yml [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+          name: docs
+          path: docs/
+      - name: Install prerequisites
+        shell: bash
+        run: |
+          set -eux
+          powershell -command "& { iwr https://github.com/ldc-developers/ldc/releases/download/v$HOST_LDC_VERSION/ldc2-$HOST_LDC_VERSION-windows-multilib.7z -OutFile ldc.7z }"
+          7z x ldc.7z
+          mv ldc2-$HOST_LDC_VERSION-windows-multilib ldc2
+          powershell -command "& { iwr http://downloads.dlang.org/other/dm857c.zip -OutFile dmc.zip }"
```
Comment (ibuclaw) [inline .github/workflows/build_windows.yml:126]: https

**Review by ibuclaw — CHANGES_REQUESTED** [review]

**Review by ibuclaw — APPROVED** [review]

**Comment by ibuclaw** [comment]
> Hmm, CI ran 2 months ago: https://github.com/dlang/installer/actions/workflows/build_windows.yml - not anymore apparently?

Not patient enough for the trigger to take effect. ;-)

**Pull request merged**